src/plots/linePlot.ts

```diff
@@ -425,6 +449,71 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+
+      let scaledXAccessor = Plot._scaledAccessor(this.x());
+      let scaledYAccessor = Plot._scaledAccessor(this.y());
+
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
```

Comment:
This looks like a piece of code that would be more clear + miniscule more performant if it was placed first
```typescript
if (indices.length === 0) {
  return [];
}
```